src/components/main-view/main-view.jsx

```diff
@@ -1,155 +1,148 @@
+            <Route path="*" element={<Navigate to="/login" />} />
+          </>
+        ) : (
+          <>
+            <Route
+              path="/movies"
+              element={
+                <>
+                  <Row>
+                    {movies.map((movie) => (
+                      <Col
+                        className="mb-5"
+                        key={movie.id}
+                        xs={12}
+                        sm={6}
+                        md={4}
+                        lg={3}
+                      >
+                        <MovieCard movie={movie} />
+                      </Col>
+                    ))}
+                  </Row>
+                  <button onClick={handleLogout}>Logout</button>
+                </>
+              }
+            />
 
-        <Row className="justify-content-md-center">
-          <Col md={8} style={{ border: "1px solid black" }}>
-            <MovieView
-              movie={selectedMovie}
-              onBackClick={() => setSelectedMovie(null)}
+            <Route
+              path="/movies/:movieId"
+              element={
+                movies.length === 0 ? (
+                  <div>Loading...</div>
+                ) : (
+                  <MovieView
+                    movies={movies} // pass movies array as prop
+                    getSimilarMovies={getSimilarMovies} // pass getSimilarMovies function as prop
+                    user={user} //TODO
+                    setUser={setUser} //TODO
+                    token={token} //TODO
+                  />
+                )
+              }
             />
-          </Col>
-        </Row>
-        <h3>Similar Movies</h3>
-        <Row className="similar-movies">
-          {similarMovies.map((movie) => (
-            <Col key={movie.id} xs={12} sm={6} md={4} lg={3}>
-              <MovieCard
-                key={movie.id}
-                movie={movie}
-                onMovieClick={(newSelectedMovie) =>
-                  setSelectedMovie(newSelectedMovie)
-                }
-              />
-            </Col>
-          ))}
-        </Row>
-      </>
-    );
-  }
 
-  if (movies.length === 0)
-    return <div className="main-view">The list is empty!</div>;
-  //main movie list view
-  return (
-    <>
-      <Row>
-        {movies.map((movie) => (
-          <Col className="mb-5" key={movie.id} xs={12} sm={6} md={4} lg={3}>
-            <MovieCard
-              movie={movie}
-              onMovieClick={(newSelectedMovie) =>
-                setSelectedMovie(newSelectedMovie)
+            <Route
+              path="/profile"
+              element={
+                <ProfileView
+                  user={user}
+                  movies={movies}
+                  token={token}
+                  setUser={setUser}
+                  onUserUpdate={handleUserUpdate} // Make sure this function is passed
+                  // onUserDeregister={handleUserDeregister}
```

Comment:
I don't believe you have defined the `handleUserDeregister` function, but you still use the `onUserDeregister` prop in the profile view component.  Please define it, and the idea was to clear the local storage after a user deregisters.

Reply:
You can also reuse the already defined `handleLogout` function like this.
```suggestion
                  onUserDeregister={handleLogout}
```

Reply:
I would like some more clarification on this. Is my onUserUpdate or handleUserUpdate not necessary since they are not defined? 
I made the change you suggested   ``` onUserDeregister={handleLogout}```

Reply:
@cinchronicity `handleUserUpdate` is defined, but it's not used after passing into the `ProfileView` Component. It's necessary but it's a function that you would call after successfully updating a users information to set the value values of emails or username to the state.